g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/utils/S3Util.java

```diff
@@ -88,18 +87,16 @@ public Locker(String basePath, SingleComponentDTO compDTO) {
 		}
 
 		public boolean lockFile() {
-			long endTime = System.currentTimeMillis() + waitToLock;
 			String content = Double.toString(random.nextDouble());
-
+             int time=0;
 			do {
-				if (!waitLockfileDisappeared(endTime)) {
+				if (!waitLockfileDisappeared()) {
 					return false;
 				}
 
 				// Write the lock file
 				s3Client.putObject(this.key, content);
 				TimeUtils.sleep(waitS3Operation); // Wait for a while to let S3 finish writing.
-
 				// Check file content is correct
 				try {
 					if (content.equals(s3Client.readObject(this.key))) {
```

Comment:
Suggest to use version id to determine that lock file is written successfully.
The last version is the most early version written. 

				PutObjectResult putResult = s3Client.putObject(this.key, "");

				// Check file is written successfully
				try {
					List<S3VersionSummary> summaries = s3Client.getS3Client().listVersions(config.getBucketName(), this.key).getVersionSummaries();
					S3VersionSummary summary = summaries.get(summaries.size()-1);
					if (putResult.getVersionId().equals(summary.getVersionId())) {

Reply:
done